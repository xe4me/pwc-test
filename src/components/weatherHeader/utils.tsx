import format from 'date-fns/format';

export const formatTodayDate = () => format(new Date(), 'cccc, MMMM do');
