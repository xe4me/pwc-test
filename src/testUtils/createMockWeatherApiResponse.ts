export const createMockWeatherApiResponse = (city_name: string) => ({
    city_name,
    data: [
      {
        wind_cdir: 'W',
        pres: 1014,
        high_temp: 20.2,
        wind_gust_spd: 5.9,
        wind_spd: 3.1,
        dewpt: 5.9,
        weather: {
          icon: 'c02d',
          code: 801,
          description: 'Few clouds',
        },
        wind_dir: 266,
        precip: 0,
        low_temp: 9.3,
        max_temp: 20.2,
        datetime: '2022-06-25',
        temp: 14.9,
        min_temp: 9.9,
      },
    ],
  });
