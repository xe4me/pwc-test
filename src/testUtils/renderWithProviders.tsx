import { render } from '@testing-library/react';
import { ReactElement } from 'react';
import { ThemeProvider } from '@mui/material';
import theme from '../theme';

export const renderWithProviders = (element: ReactElement) =>
  render(<ThemeProvider theme={theme}>{element}</ThemeProvider>);
