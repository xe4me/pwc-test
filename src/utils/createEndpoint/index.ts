import { envs } from 'constants/envs';

/*
 * @Function createEndpoint
 * This function will add the BASE URL and the authentication key to the given endpoint
 * Note that this function is very limited and does not consider whether the endpoint has "$" or not
 * */
export const createEndpoint = (endpoint: string) =>
  `${envs.weatherApiBaseUrl}${endpoint}&key=${envs.authKey}`;
