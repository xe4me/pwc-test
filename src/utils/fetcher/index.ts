/*
 * @Function fetcher
 * This is a wrapper around the browser's fetch function, required by SNW library
 * */
// @ts-ignore
export const fetcher = (...args) => fetch(...args).then((res) => res.json());
