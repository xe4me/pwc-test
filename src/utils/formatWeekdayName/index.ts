import format from 'date-fns/format';
import isToday from 'date-fns/isToday';

/*
* @Function formatWeekdayName
* Will format the given date to one of these values
* [
    "Sunday",
    "Monday",
    "Tuesday",
    "Wednesday",
    "Thursday",
    "Friday",
    "Saturday",
  ]
  * If the given date is current day, it will return Today
* */
export const formatWeekdayName = (datetime: Date) =>
  isToday(datetime) ? 'Today' : format(datetime, 'cccc');
