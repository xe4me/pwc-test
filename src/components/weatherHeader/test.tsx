import React from 'react';
import { screen } from '@testing-library/react';
import { renderWithProviders } from 'testUtils/renderWithProviders';
import { axe } from 'jest-axe';
import { createMockWeatherApiResponse } from 'testUtils/createMockWeatherApiResponse';
import { WeatherHeader } from './index';
import { formatTodayDate } from './utils';

const renderComponent = () => {
  const data = createMockWeatherApiResponse('Sydney');
  return renderWithProviders(
    <WeatherHeader city={data.city_name} forecast={data.data[0]} />
  );
};
describe('<WeatherHeader>', () => {
  describe('Snapshot and accessibility', () => {
    it('should match with the snapshot', () => {
      const { baseElement } = renderComponent();
      expect(baseElement).toMatchSnapshot();
    });
    it('should pass accessibility tests', async () => {
      const html = renderComponent();
      expect(await axe(html.container)).toHaveNoViolations();
    });
  });
  describe('Visible Elements', () => {
    it('should render the given city', () => {
      renderComponent();
      expect(screen.queryByText('Sydney')).toBeInTheDocument();
    });
    it('should render formatted today', () => {
      renderComponent();
      expect(screen.queryByText(formatTodayDate())).toBeInTheDocument();
    });
  });
});
