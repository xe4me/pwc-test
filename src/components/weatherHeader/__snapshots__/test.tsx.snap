// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeatherHeader> Snapshot and accessibility should match with the snapshot 1`] = `
<body>
  <div>
    <h2
      class="MuiTypography-root MuiTypography-h2 MuiTypography-gutterBottom css-qvi7wk-MuiTypography-root"
    >
      Sydney
    </h2>
    <h3
      class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-1dcqdis-MuiTypography-root"
    >
      Saturday, June 25th
    </h3>
    <h3
      class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-1dcqdis-MuiTypography-root"
    >
      Overcast
    </h3>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-a5rdam-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-6 css-1lxbjw-MuiGrid-root"
      >
        <img
          alt="Today's weather image"
          class="css-g8qt4o"
          src="https://www.weatherbit.io/static/img/icons/c02d.png"
        />
        <div
          class="css-1yp4ln"
        >
          <span
            class="MuiTypography-root MuiTypography-h2 MuiTypography-gutterBottom css-7j95bm-MuiTypography-root"
          >
            14.9
          </span>
          <span
            class="MuiTypography-root MuiTypography-h5 css-fru0ub-MuiTypography-root"
          >
            â„ƒ
          </span>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
      >
        <table>
          <tbody>
            <tr>
              <td>
                <span
                  class="MuiTypography-root MuiTypography-h5 css-fru0ub-MuiTypography-root"
                >
                  Precipitation:
                   
                </span>
                <span
                  class="MuiTypography-root MuiTypography-h5 css-11zemyq-MuiTypography-root"
                >
                  0
                </span>
              </td>
            </tr>
            <tr>
              <td>
                <span
                  class="MuiTypography-root MuiTypography-h5 css-fru0ub-MuiTypography-root"
                >
                  Humidity:
                   
                </span>
                <span
                  class="MuiTypography-root MuiTypography-h5 css-11zemyq-MuiTypography-root"
                >
                  5.9
                </span>
              </td>
            </tr>
            <tr>
              <td>
                <span
                  class="MuiTypography-root MuiTypography-h5 css-fru0ub-MuiTypography-root"
                >
                  Wind:
                   
                </span>
                <span
                  class="MuiTypography-root MuiTypography-h5 css-11zemyq-MuiTypography-root"
                >
                  3.1
                   kph 
                  W
                </span>
              </td>
            </tr>
            <tr>
              <td>
                <span
                  class="MuiTypography-root MuiTypography-h5 css-fru0ub-MuiTypography-root"
                >
                  Pollen Count:
                   
                </span>
                <span
                  class="MuiTypography-root MuiTypography-h5 css-11zemyq-MuiTypography-root"
                >
                  14.9
                </span>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</body>
`;
